prac_07/project_management.py

```diff
@@ -0,0 +1,138 @@
+            save_projects(filename, projects)
+
+        elif choice == 'd':
+            display_projects(projects)
+
+        elif choice == 'f':
+            start_date = datetime.datetime.strptime(input("Show projects that start after date (dd/mm/yy): "), "%d/%m/%Y").date()
+            filter_projects_by_date(projects, start_date)
+
+        elif choice == 'a':
+            add_new_project(projects)
+
+        elif choice == 'u':
+            update_project(projects)
+
+        elif choice == 'q':
+            break
+
+        else:
+            print("Invalid choice. Please try again.")
+
+    print("Thank you for using custom-built project management software.")
+
+
+def load_projects(filename):
+    projects = []
+    try:
+        with open(filename, "r") as file:
+            lines = file.readlines()
+            if lines and lines[0].strip() == "Name\tStart Date\tPriority\tCost Estimate\tCompletion Percentage":  # Check if the first line is the header
+                lines = lines[1:]  # Skip the header line
+            for line in lines:
+                project_data = line.strip().split("\t")  # Use tabs ("\t") as the separator
+                if len(project_data) == 5:  # Ensure all fields are present
+                    name = project_data[0]
+                    start_date = datetime.datetime.strptime(project_data[1], "%d/%m/%Y").date()
+                    priority = int(project_data[2])
+                    estimate = float(project_data[3])
+                    completion = int(project_data[4])
+                    projects.append(Project(name, start_date, priority, estimate, completion))
+                else:
+                    print(f"Error reading project data: {line.strip()}")
+    except FileNotFoundError:
+        print("File not found. Starting with an empty project list.")
+    return projects
+
+
+def save_projects(filename, projects):
+    with open(filename, "w") as file:
+        for project in projects:
+            file.write(f"{project.name},{project.start_date.strftime('%d/%m/%Y')},{project.priority},"
+                       f"{project.estimate},{project.completion}\n")
+
+
+def display_projects(projects):
+    incomplete_projects = [project for project in projects if not project.is_completed()]
+    completed_projects = [project for project in projects if project.is_completed()]
+
+    print("Incomplete projects:")
+    for project in sorted(incomplete_projects, key=lambda x: x.priority):
+        print(" ", project)
+    print("Completed projects:")
+    for project in sorted(completed_projects, key=lambda x: x.priority):    # Decided to use the lambda function for
+        print(" ", project)                                                 # sorting to make easier.
+
+
+def filter_projects_by_date(projects, start_date):
+    filtered_projects = [project for project in projects if project.start_date > start_date]
+    for project in sorted(filtered_projects, key=lambda x: x.start_date):
```

Comment:
Could you avoid creating a new list here by using this approach instead (and maybe using `filter_date` as the variable name for the filter, rather than `start_date`)?
```suggestion
    for project in sorted(projects, key=lambda x: x.start_date):
         if project.start_date > start_date:
              print(project)
```